src/deploy/functions/params.ts

```diff
@@ -66,29 +54,22 @@ export function resolveString(
   from: string | build.Expression<string>,
   paramValues: Record<string, ParamValue>
 ): string {
-  if (!isCEL(from)) {
-    return from;
-  }
   let output = from;
-  const paramCapture = /{{ params\.(\w+) }}/g;
+  const celCapture = /{{ (.+?) }}/;
+  let infiniteLoopGuard = 0;
   let match: RegExpMatchArray | null;
-  while ((match = paramCapture.exec(from)) != null) {
-    const referencedParamValue = paramValues[match[1]];
-    if (!referencedParamValue.legalString) {
+  while ((match = celCapture.exec(output)) != null) {
+    infiniteLoopGuard += 1;
+    if (infiniteLoopGuard > 100) {
```

Comment:
q: why is it that `resolveString` is the only param that resolves nested CEL expressions?

Reply:
This is interpolation, not nesting. The revised version of this function should make that more clear.